stock_picking_customer_ref/models/stock_picking.py

```diff
@@ -11,13 +10,16 @@ class StockPicking(models.Model):
 
     @api.depends('group_id', 'sale_id.client_order_ref')
     def _compute_client_order_ref(self):
-        for picking in self:
-            if picking.group_id:
-                cond = [('procurement_group_id', '=', picking.group_id.id)]
-                sale = self.env['sale.order'].search(cond, limit=1)
-                if sale:
-                    picking.client_order_ref = sale.client_order_ref
+        stock_pickings = self.filtered(
+            lambda stock_picking: stock_picking.group_id)
+        for picking in stock_pickings:
+            cond = [('procurement_group_id', '=', picking.group_id.id)]
+            sale = self.env['sale.order'].search(cond, limit=1)
+            if not sale:
+                continue
+            picking.client_order_ref = sale.client_order_ref
 
-    client_order_ref = fields.Char(string="Sale Reference/Description",
-                                   compute="_compute_client_order_ref",
-                                   store=True)
+    client_order_ref = fields.Char(
+        string="Sale Reference/Description",
```

Comment:
Better to use `Customer Reference` as a string instead of `Sale Reference/Description` here (and at all other places too).

Ps: SO number will be already there in origin field
![screenshot 11](https://user-images.githubusercontent.com/36026345/50371256-f6055780-05dc-11e9-8bd8-bfd895375beb.png)

Reply:
> Better to use `Customer Reference` as a string instead of `Sale Reference/Description` here (and at all other places too).
> 
> Ps: SO number will be already there in origin field
> ![screenshot 11](https://user-images.githubusercontent.com/36026345/50371256-f6055780-05dc-11e9-8bd8-bfd895375beb.png)

@Swapnesh-SerpentCS  I doubt about it, because Source Document has already the value of Sale order, and the purpose of the module is to get the Sale order client reference field into Delivery order

Reply:
@elicoidal Can you please suggest what should be the String/caption for the field. as we doubt for the same. as per @Swapnesh-SerpentCS it should be Customer Reference (Same as Sale order) and according to my it should have Sale Customer Reference. as it is related field from Sale order.

Reply:
@elicoidal  Thank you for your feedback. Changes are done as per you feedback.